### DISCUSSION TITLE
Tabbed content in Workbench?

### DISCUSSION THREAD

**Discussion by deppen8** [carpentries/workbench#5] [Q&A] [answered]
Not sure if anyone is monitoring these Discussions, but I am curious if the Workbench setup will allow us to use Rmarkdown features like tabbed content, a la https://bookdown.org/yihui/rmarkdown-cookbook/html-tabs.html.

This might be an especially useful feature for the `shell-novice` lesson I help maintain. We sometimes struggle to handle the slight variations in commands between different operating systems and this could be a succinct, uncomplicated way to include those.

(We actually have this type of tab in our current [Setup page](https://raw.githubusercontent.com/swcarpentry/shell-novice/gh-pages/setup.md), but it requires quite a bit of HTML that I'd rather not have to maintain everywhere.)

**Comment by zkamvar** [comment, marked as answer]
(n.b. I've transferred this discussion to the workbench repository because I want to try to centralise these discussions)

## Short Answer

It's not available yet, but we are planning on it (see https://github.com/carpentries/varnish/issues/32)! 

In the meantime, you can use `solution` blocks for dropdowns appended to a discussion block [as shown in the markdown template](https://github.com/carpentries/workbench-template-md/blob/6182dbd68dca5f6c2e146bdd923a3f3117d789a9/learners/setup.md?plain=1#L12-L48):

```markdown

## Data

## Software

::: discussion

### Software Setup

[broad instructions]

:::

::: solution

### Windows

[something about GitBash]

:::

::: solution

### MacOS

[Something about Terminal.app]

:::

::: solution

### Linux

[Something about assuming Ubuntu]

:::
```

This is related to https://github.com/carpentries/sandpaper/issues/28

## Longer answer

It's not available because we still need to:

1. define the tabset behaviour in CSS and JS
2. make sure the elements meets [WCAG 2.1 AA accessibility standards](https://www.w3.org/TR/WCAG21/)
3. explicitly test accessibility (with help from our Accessibility Coordinator)

Funny enough, the initial Figma renders of the setup page actually _included_ tabbed content for the setup:

![screenshot from Figma showing a setup callout block showing tabbed content titled "windows", "mac", and "linux"](https://user-images.githubusercontent.com/3639446/167164858-6af15842-53b3-46e2-869c-dd12abab873d.png)

The problem is that the contractors who wrote the HTML, JS, and CSS missed this content when they wrote it in December 2021 and we needed to keep moving :(

I will open an issue for this in {varnish}, where we will be able to incorporate this feature.

**Reply by deppen8** [reply]
Thanks for the detail @zkamvar . I'll probably hold off until it is officially supported, but excited to hear that it is in the conversation.
